vortex-array/src/compute/conformance/cast.rs

```diff
@@ -106,436 +93,139 @@ fn test_cast_from_null(array: &dyn Array) {
-            assert_eq!(back_to_non_nullable.len(), array.len());
-
-            // Verify values are unchanged
-            for i in 0..array.len().min(10) {
-                assert_eq!(array.scalar_at(i), back_to_non_nullable.scalar_at(i));
-            }
-        }
-    }
-}
-
-fn test_cast_nullability_changes(array: &dyn Array, nullable_version: &DType) {
-    // Test casting to nullable version
-    if array.dtype().nullability() == Nullability::NonNullable {
-        let result = cast(array, nullable_version).vortex_unwrap();
-        assert_eq!(result.len(), array.len());
-        assert_eq!(result.dtype(), nullable_version);
-
-        // IMPORTANT: Nullability casting should preserve the encoding
-        assert_eq!(
-            result.encoding().id(),
-            array.encoding().id(),
-            "Nullability cast should preserve encoding"
-        );
+        let non_nullable = cast(array, &array.dtype().as_nonnullable())
+            .vortex_expect("arrays without nulls can cast to non-nullable");
+        assert_eq!(non_nullable.dtype(), &array.dtype().as_nonnullable());
+        assert_eq!(non_nullable.len(), array.len());
 
-        // Values should be unchanged
         for i in 0..array.len().min(10) {
-            assert_eq!(array.scalar_at(i), result.scalar_at(i),);
+            assert_eq!(array.scalar_at(i), non_nullable.scalar_at(i));
         }
-    }
-}
-
-fn test_cast_nullability_changes_primitive(
-    array: &dyn Array,
-    ptype: PType,
-    nullability: Nullability,
-) {
-    // Test casting to nullable version
-    if nullability == Nullability::NonNullable {
-        let nullable_dtype = DType::Primitive(ptype, Nullability::Nullable);
-        let result = cast(array, &nullable_dtype).vortex_unwrap();
-        assert_eq!(result.len(), array.len());
-        assert_eq!(result.dtype(), &nullable_dtype);
 
-        // IMPORTANT: Nullability casting should preserve the encoding
-        assert_eq!(
-            result.encoding().id(),
-            array.encoding().id(),
-            "Nullability cast should preserve encoding"
-        );
+        let back_to_nullable = cast(&non_nullable, array.dtype())
+            .vortex_expect("non-nullable arrays can cast to nullable");
+        assert_eq!(back_to_nullable.dtype(), array.dtype());
+        assert_eq!(back_to_nullable.len(), array.len());
 
-        // Values should be unchanged
         for i in 0..array.len().min(10) {
-            assert_eq!(array.scalar_at(i), result.scalar_at(i),);
+            assert_eq!(array.scalar_at(i), back_to_nullable.scalar_at(i));
         }
-    }
-
-    // Test casting from nullable to non-nullable (only if no nulls present)
-    if nullability == Nullability::Nullable {
-        // Try to cast to non-nullable and see if it succeeds
-        let non_nullable_dtype = DType::Primitive(ptype, Nullability::NonNullable);
-        if let Ok(result) = cast(array, &non_nullable_dtype) {
```

Comment:
This seems to be a dupe of the previously commented `if let` (which also didn't verify all-valid arrays were castable to non-nullable)